presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PushdownSubfields.java

```diff
@@ -601,9 +777,39 @@ public Void visitSpecialForm(SpecialFormExpression specialForm, Context context)
                 return null;
             }
 
+            /**
+             *
+             * Adds lambda subfields from the context to the list of the required subfields of the field/subscript/subfield provided in parameter 'input'. This function should be
+             * invoked
+             * once we reached leaf node while visiting the expression tree. Effectively, it prunes all unaccessed subfields of the 'input'.
+             *
+             * @param context - SubfieldExtractor context
+             * @param input - input field, subscript, or subfield, for which lambda subfields were collected.
+             */
+            @Nullable
+            private void addRequiredLambdaSubfields(Context context, Subfield input)
+            {
+                if (!isPushDownSubfieldsFromLambdasEnabled || !context.getLambdaSubfields().isPresent()) {
```

Comment:
we shouldn't need to check that here.  The code should be structured such that it's not possible to get here (or to get here and cause wrong results) if this field is false. ideally we only want to have one place where we are gating based on this property.

Reply:
Removed